core/keyboard_nav/navigation.js

```diff
@@ -644,15 +601,37 @@ Blockly.navigation.getSourceBlock_ = function(node) {
   }
 };
 
+/**
+ * Gets the top node on a block.
+ * This is either the previous connection, output connection or the block.
+ * @param {Blockly.Block} block The block to find the top most AST node on.
+ * @return {Blockly.ASTNode} The AST node holding the top most node on the
+ *     block.
+ * @package
+ */
+Blockly.navigation.getTopNode = function(block) {
+  var prevConnection = block.previousConnection;
+  var outConnection = block.outputConnection;
+  var topConnection = prevConnection ? prevConnection : outConnection;
```

Comment:
These three lines should just be:
var topConnection = block.previousConnection || block.outputConnection;